gems/sorbet-runtime/lib/types/types/simple.rb

```diff
@@ -16,7 +18,7 @@ def name
       #
       # `name` isn't normally a hot path for types, but it is used in initializing a T::Types::Union,
       # and so in `T.nilable`, and so in runtime constructions like `x = T.let(nil, T.nilable(Integer))`.
-      @name ||= @raw_type.name.freeze
+      @name ||= NAME_METHOD.bind_call(@raw_type).freeze
```

Comment:
Since we make concessions to other pre-Ruby 2.7 versions elsewhere in `sorbet-runtime`, we should do the same here by using `bind(@raw_type).call` on those versions.